pkg/speccheck/speccheck.go

```diff
@@ -28,10 +29,28 @@ func New(kts ...map[schema.GroupKind]ObjectProcessor) *SpecCheck {
 		}
 	}
 	return &SpecCheck{
+		Store:      store,
 		KnownTypes: kt,
 	}
 }
 
+// BeforeCreate pre-processes object specification and returns an updated version.
+func (sc *SpecCheck) BeforeCreate(logger *zap.Logger, spec *unstructured.Unstructured) (*unstructured.Unstructured /*updatedSpec*/, error) {
+	processor, ok := sc.KnownTypes[spec.GroupVersionKind().GroupKind()]
```

Comment:
The removed `switch` from the `resource_sync_task.go` is this lookup now essentially (and the same lookup in `compareActualVsSpec()` below).